modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_in.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *    
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.filter.function;
+
+import java.util.List;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+
+/**
+ * The function checks whether a candidate value is contained in an arbitrary long list
+ * of user provided values.
+ * 
+ * <p>
+ * If the candidate value is found, the function returns <code>true</code>;
+ * otherwise, it returns <code>false</code>.
+ * </p>
+ * 
+ * @author Stefano Costa, GeoSolutions
+ * 
+ * 
+ * 
+ * @source $URL$
+ *
+ */
+public class FilterFunction_in extends FunctionExpressionImpl {
+
+    public static FunctionName NAME = functionName("in", "result:Boolean", "candidate:Object:1,1",
+            "v:Object:1,");
+
+    public FilterFunction_in() {
+        super(NAME);
+    }
+
+    @Override
+    public String getName() {
+        return NAME.getName();
+    }
+
+    public int getArgCount() {
+        return NAME.getArgumentCount();
+    }
+
+    public Object evaluate(Object feature) {
+        Object candidate = getExpression(0).evaluate(feature);
+
+        boolean result = false;
+
+        if (candidate != null) {
+            List<Expression> valuesToTest = (List<Expression>) getParameters().subList(1,
+                    getParameters().size());
+            for (Expression expression : valuesToTest) {
+                Object value = expression.evaluate(feature);
```

Comment:
This part is difficult as you want to use the two argument form: evaluate( feature, type ) where "type" is is candidate.getClass().

This will let you preserve (as best we can) the CQL Expressions being typless.